control-plane/tests/test_phase29_mlflow_shadow_model_registry_validation.py

```
@@ -0,0 +1,514 @@
+
+class UnexpectedRegisteredModelLookupClient(FakeMlflowClient):
+    def get_registered_model(self, name: str) -> _FakeRegisteredModel | None:
+        raise _FakeMlflowLookupError(
+            f"authentication failed while loading {name}",
+            error_code="PERMISSION_DENIED",
+        )
+
+
+class Phase29MlflowShadowModelRegistryValidationTests(ServicePersistenceTestBase):
+    def test_tracker_records_mlflow_shadow_run_and_registry_lineage(self) -> None:
+        store, service, dataset_snapshot, decided_at = self._build_shadow_dataset_snapshot()
```

Ruff (0.15.10)

[warning] Unpacked variable `service` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)